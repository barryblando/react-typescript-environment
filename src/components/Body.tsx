import React, { PureComponent } from 'react';
import { TransitionGroup, CSSTransition } from 'react-transition-group';

import styles from '../styles/index.scss';
import buttonsStyles from '../styles/button.scss';
import body from '../styles/body.scss';

const FIRST_IMAGE_ID = 1;
const SECOND_IMAGE_ID = 2;

export class Body extends PureComponent<IBody, {}> {
  _handleClick = () => {
    const { imageToShow, switchImage } = this.props;
    if (imageToShow === FIRST_IMAGE_ID) {
      return switchImage ? switchImage(SECOND_IMAGE_ID) : {};
    }
    return switchImage ? switchImage(FIRST_IMAGE_ID) : {};
  }

  _getImage = () => {
    const { imageToShow } = this.props;

    return imageToShow;
  }

  render() {
    const imageToShow = this._getImage();

    return (
      <div className={styles.body}>
        <button type='button' onClick={this._handleClick} className={buttonsStyles.button}>
          <TransitionGroup className={body.animWrap}>
            <CSSTransition classNames='mainImage' timeout={500} key={imageToShow}>
              <img className={styles.bodyImg} src={`../assets/${imageToShow}.png`} alt='main_img' />
            </CSSTransition>
          </TransitionGroup>
        </button>
      </div>
    );
  }
}

export default Body;
