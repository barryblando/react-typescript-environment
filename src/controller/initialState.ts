const initialState = {
  app: {
    appName: 'REACT_TS_ENV',
    imageToShow: 1
  }
};

export default initialState;
