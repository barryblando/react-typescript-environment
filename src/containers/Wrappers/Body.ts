import { Dispatch } from 'redux';
import { connect } from 'react-redux';
import { Body } from '../../components';
import { switchImage } from '../../modules/actions';

const mapStateToProps = (state: any) => ({
  imageToShow: state.app.imageToShow
});

const mapDispatchToProps = (dispatch: Dispatch) => ({
  switchImage: (imageID: number) => dispatch(switchImage(imageID))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Body);
