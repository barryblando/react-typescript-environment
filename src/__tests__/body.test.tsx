import React from 'react';
import { shallow } from 'enzyme';

import initialState from './mocks/body';

import { Body } from '../components/Body';

describe('<Body Component Test />', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallow(<Body {...initialState} />);
  });

  test('should render basic Body Component with an image', () => {
    expect(wrapper.find('.body').length).toBe(1);
    expect(wrapper.find('.bodyImg').length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });
});
