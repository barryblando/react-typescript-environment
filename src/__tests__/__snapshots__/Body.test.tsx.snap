// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Body Component Test /> should render basic Body Component with an image 1`] = `
<div
  className="body"
>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    <TransitionGroup
      childFactory={[Function]}
      className="animWrap"
      component="div"
    >
      <CSSTransition
        classNames="mainImage"
        timeout={500}
      >
        <img
          alt="main_img"
          className="bodyImg"
          src="../assets/undefined.png"
        />
      </CSSTransition>
    </TransitionGroup>
  </button>
</div>
`;
